### PR TITLE
#626 StatsD auto initialization

### PR DESCRIPTION
**Pull request by kfrz**
## Description of change
- add a `StatsDMetric` class, inheriting from the `Common::RedisStore` class, used to write keys to redis 
- modify `with_monitoring` lambda to utilize the `StatsDMetric` class

## Testing done
- add spec for `::Monitoring` concern
- test by inspecting with `redis-cli` and `byebug` that the metric keys are written to redis

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/common/client/concerns/monitoring.rb [hunk 1]
```diff
@@ -14,14 +14,28 @@ def with_monitoring(trace_location = 1)
...
     def increment_failure(caller, error)
       tags = ["error:#{error.class}"]
       tags << "status:#{error.status}" if error.try(:status)
-      StatsD.increment("#{self.class::STATSD_KEY_PREFIX}.#{caller}.fail", tags: tags)
+
+      increment("#{self.class::STATSD_KEY_PREFIX}.#{caller}.fail", tags: tags)
+    end
+
+    def increment(tag)
+      # TODO this would poll redis for each call of with_monitoring.
```
Comment (LindseySaari) [inline lib/common/client/concerns/monitoring.rb]: Each time we increment a metric, we will add that metric name to our running redis set correct?

File: app/workers/statsd_census_job.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  def perform
+    initialize_stats_roster unless stats_roster_is_in_redis?
+    audit_roster!
+  end
+
+  private
+
+  def initialize_stats_roster
+    Redis.current.set(stats_roster_key, 1)
```
Comment (LindseySaari) [inline app/workers/statsd_census_job.rb]: A Redis SET might be our best option for the initialized statsD key storage “It is possible to add, remove, and test for existence of members in O(1) (constant time regardless of the number of elements contained inside the Set).” https://www.tutorialspoint.com/redis/redis_sets. Sorted Sets have the notion of scores, which doesn’t apply to us. https://redis.io/topics/data-types-intro. Redis SETS don’t allow repeated members

File: app/workers/statsd_census_job.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def stats_roster_is_in_redis?
```
Comment (LindseySaari) [inline app/workers/statsd_census_job.rb]: Is this where we want to grab our SET of statsD keys from redis or check if the SET exists?

File: spec/lib/common/client/concerns/monitoring_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Common::Client::Monitoring, type: :model do
```
Comment (kfrz) [inline spec/lib/common/client/concerns/monitoring_spec.rb]: This should move bucko

File: Guardfile [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+directories(%w[app lib config spec].select { |d| Dir.exist?(d) ? d : UI.warning("Directory #{d} does not exist") })
```
[the remark this answers is not in the document]
Comment (kfrz) [reply Guardfile]: Whoops - didn't mean to commit this. Thanks

File: config/environments/production.rb [hunk 6]
```diff
@@ -18,6 +18,9 @@
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
 
+  # Adding 'lib/' to eager load so we can automatically increment StatsD totals initializers
+  config.eager_load_paths << Rails.root.join('lib')
```
[the remark this answers is not in the document]
Comment (kfrz) [reply config/environments/production.rb]: This is actually a relic of a previous approach, not needed to eager load with the latest updates

File: Guardfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (kfrz) [reply Guardfile]: A relic of my dev environment, removed.

**Comment by kfrz** [comment]
```bash
13:19:32 oddball/department-of-veterans-affairs/vets-api  master [$]    ruby-2.4.5@vets-api                                                               
➔ ab -n 5000 -c 50 localhost:3000/v0/search                                                                                                                 
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>                                                                                                     
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/                                                                                    
Licensed to The Apache Software Foundation, http://www.apache.org/                                                                                          
                                                                                                                                                            
Benchmarking localhost (be patient)                                                                                                                         
Completed 500 requests                                                                                                                                      
Completed 1000 requests                                                                                                                                     
Completed 1500 requests                                                                                                                                     
Completed 2000 requests                                                                                                                                     
Completed 2500 requests                                                                                                                                     
Completed 3000 requests                                                                                                                                     
Completed 3500 requests                                                                                                                                     
Completed 4000 requests                                                                                                                                     
Completed 4500 requests                                                                                                                                     
Completed 5000 requests                                                                                                                                     
Finished 5000 requests                                                                                                                                      
                                                                                                                                                            
                                                                                                                                                            
Server Software:                                                                                                                                            
Server Hostname:        localhost                                                                                                                           
Server Port:            3000                                                                                                                                
                                                                                                                                                            
Document Path:          /v0/search                                                                                                                          
Document Length:        8685 bytes                                                                                                                          
                                                                                                                                                            
Concurrency Level:      50                                                                                                                                  
Time taken for tests:   282.824 seconds                                                                                                                     
Complete requests:      5000                                                                                                                                
Failed requests:        0                                                                                                                                   
Non-2xx responses:      5000                                                                                                                                
Total transferred:      45080000 bytes                                                                                                                      
HTML transferred:       43425000 bytes                                                                                                                      
Requests per second:    17.68 [#/sec] (mean)                                                                                                                
Time per request:       2828.244 [ms] (mean)                                                                                                                
Time per request:       56.565 [ms] (mean, across all concurrent requests)                                                                                  
Transfer rate:          155.66 [Kbytes/sec] received                                                                                                        
                                                                                                                                                            
Connection Times (ms)                                                                                                                                       
              min  mean[+/-sd] median   max                                                                                                                 
Connect:        0    0   0.2      0       5                                                                                                                 
Processing:   301 2806 632.1   2720    6312                                                                                                                 
Waiting:      301 2804 630.8   2718    6312                                                                                                                 
Total:        302 2806 632.1   2720    6312                                                                                                                 
                                                                                                                                                            
Percentage of the requests served within a certain time (ms)                                                                                                
  50%   2720                                                                                                                                                
  66%   2963                                                                                                                                                
  75%   3114                                                                                                                                                
  80%   3215                                                                                                                                                
  90%   3522                                                                                                                                                
  95%   3855                                                                                                                                                
  98%   4663                                                                                                                                                
  99%   5085                                                                                                                                                
 100%   6312 (longest request) 
```


```bash
13:24:58 oddball/department-of-veterans-affairs/vets-api  kfrz/626-statsd-auto-initialization [$⇣]    ruby-2.4.5@vets-api                                 
➔ ab -n 5000 -c 50 localhost:3000/v0/search                                                                                                                 
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>                                                                                                     
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/                                                                                    
Licensed to The Apache Software Foundation, http://www.apache.org/                                                                                          
                                                                                                                                                            
Benchmarking localhost (be patient)                                                                                                                         
Completed 500 requests                                                                                                                                      
Completed 1000 requests                                                                                                                                     
Completed 1500 requests                                                                                                                                     
Completed 2000 requests                                                                                                                                     
Completed 2500 requests                                                                                                                                     
Completed 3000 requests                                                                                                                                     
Completed 3500 requests                                                                                                                                     
Completed 4000 requests                                                                                                                                     
Completed 4500 requests                                                                                                                                     
Completed 5000 requests                                                                                                                                     
Finished 5000 requests                                                                                                                                      
                                                                                                                                                            
                                                                                                                                                            
Server Software:                                                                                                                                            
Server Hostname:        localhost                                                                                                                           
Server Port:            3000                                                                                                                                
                                                                                                                                                            
Document Path:          /v0/search                                                                                                                          
Document Length:        2316 bytes                                                                                                                          
                                                                                                                                                            
Concurrency Level:      50                                                                                                                                  
Time taken for tests:   191.272 seconds                                                                                                                     
Complete requests:      5000                                                                                                                                
Failed requests:        0                                                                                                                                   
Total transferred:      13495000 bytes                                                                                                                      
HTML transferred:       11580000 bytes                                                                                                                      
Requests per second:    26.14 [#/sec] (mean)                                                                                                                
Time per request:       1912.719 [ms] (mean)                                                                                                                
Time per request:       38.254 [ms] (mean, across all concurrent requests)                                                                                  
Transfer rate:          68.90 [Kbytes/sec] received                                                                                                         
                                                                                                                                                            
Connection Times (ms)                                                                                                                                       
              min  mean[+/-sd] median   max                                                                                                                 
Connect:        0    0   0.2      0       5                                                                                                                 
Processing:   454 1901 631.3   1701    5073                                                                                                                 
Waiting:      454 1900 629.9   1700    5073                                                                                                                 
Total:        455 1902 631.3   1701    5074                                                                                                                 
                                                                                                                                                            
Percentage of the requests served within a certain time (ms)                                                                                                
  50%   1701                                                                                                                                                
  66%   1885                                                                                                                                                
  75%   2054                                                                                                                                                
  80%   2187                                                                                                                                                
  90%   2803                                                                                                                                                
  95%   3376                                                                                                                                                
  98%   3885                                                                                                                                                
  99%   4153                                                                                                                                                
 100%   5074 (longest request)  
```

**Comment by kfrz** [comment]
```

  5) V0::SessionsController when logged in POST saml_callback changing multifactor has a cookie, which includes the testing values
     # Temporarily skipped with xit
     # ./spec/controllers/v0/sessions_controller_spec.rb:409

     Common::Client::SecurityError:
       http client needs cookies stripped
nt::Secu
       expected Common::Client::Errors::ClientError, got #<Common::Client::SecurityError: h...............................................................................................................................................................No value provided for required arguments 'name'
...........................................................................................................................................................................................................................................................................*....................................................................................................................................................................................................................VCR: got `Encoding::UndefinedConversionError: "\xE2" from ASCII-8BIT to UTF-8` while trying to encode the ASCII-8BIT request body to the original body encoding (UTF-8). Consider using the `:preserve_exact_body_bytes` option to work around this.
VCR: got `Encoding::UndefinedConversionError: "\xE2" from ASCII-8BIT to UTF-8` while trying to encode the ASCII-8BIT request body to the original body encoding (UTF-8). Consider using the `:preserve_exact_body_bytes` option to work around this.
...........................................................................................      create  modules/foo/lib/foo/engine.rb
      create  modules/foo/lib/foo/version.rb
      create  modules/foo/lib/tasks/foo_tasks.rake
      create  modules/foo/lib/foo.rb
....      create  modules/foo/app/controllers/foo/v0/foo_controller.rb
      create  modules/foo/app/controllers/foo/application_controller.rb
      create  modules/foo/app/models/foo/resource.rb
      create  modules/foo/app/serializers/foo/foo_serializer.rb
      create  modules/foo/app/services/foo/configuration.rb
      create  modules/foo/app/services/foo/service.rb
......      create  modules/foo/bin/rails
       chmod  modules/foo/bin/rails
      create  modules/foo/config/routes.rb
      create  modules/foo/foo.gemspec
      create  modules/foo/Rakefile
      create  modules/foo/Gemfile
      create  modules/foo/README.rdoc
.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................FF.......... -S -m "Add cookie stripper"
Running pre-commit hooks
Run InternalAddresses.............................[InternalAddresses] OK
Check for trailing whitespace....................[TrailingWhitespace] OK
Check for "token" strings.....................................[FixMe] OK
Analyze with RuboCop........................................[RuboCop] OK

✓ All pre-commit hooks passed

Running commit-msg hooks
Check for trailing periods in subject................[TrailingPeriod] OK
Check subject capitalization.....................[CapitalizedSubject] OK
Check text width..........................................[TextWidth] OK
Check subject line................................[SingleLineSubject] OK

✓ All commit-msg hooks passed

[kfrz/626-statsd-auto-initialization e4c61a6e9] Add cookie stripper
 1 file changed, 1 insertion(+)
.FF..#<Commo Common::Client::SecurityError:
       http client needs cookies stripped
lspec8b8a6e6ca HEAD@{32}: reset: moving to HEAD
AD@{101}: checkout: moving from kfrz/mitigate-appeals-enoent to master
64825e78d HEAD@{102}: commit: Stringify
 SecurityError, .........................FF...........



pected Common::Client::Errors::Clien_error(Common::Client::Errors::ClientError)
       expected Common::Client::Errors:: expected nil to evaluate to true
#<VCR::Errors::UnhandledHTTPRequestError:

       =========================================================...n/ignore-request
       ================================================================================

       > with backtrace:
 Specs::Common::Client::TestService.new.send(:request, :get, '', nil) }.to raise_error(
         Common::Client::SecurityError
       )
ce.new.sendspec.  expected Common::Client::SecurityError, got #<VCR::Errors::UnhandledHTTPRequestError:

       =========================================================...n/ignore-request
       ================================================================================

       > with backtrace:
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/vcr-3.0.3/lib/vcr/request_handler.rb:97:in `on_unhandled_request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/vcr-3.0.3/lib/vcr/library_hooks/webmock.rb:129:in `on_unhandled_request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/vcr-3.0.3/lib/vcr/request_handler.rb:24:in `handle'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/vcr-3.0.3/lib/vcr/library_hooks/webmock.rb:144:in `block in <module:WebMock>'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/stub_registry.rb:28:in `block in register_global_stub'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/request_pattern.rb:40:in `matches?'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/stub_registry.rb:58:in `block in request_stub_for'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/stub_registry.rb:57:in `each'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/stub_registry.rb:57:in `detect'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/stub_registry.rb:57:in `request_stub_for'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/stub_registry.rb:50:in `response_for_request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/http_lib_adapters/httpclient_adapter.rb:187:in `block in webmock_responses'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/http_lib_adapters/httpclient_adapter.rb:59:in `do_get'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/webmock-3.7.2/lib/webmock/http_lib_adapters/httpclient_adapter.rb:47:in `do_get_block'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/httpclient-2.8.3/lib/httpclient.rb:1019:in `block in do_request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/httpclient-2.8.3/lib/httpclient.rb:1133:in `protect_keep_alive_disconnected'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/httpclient-2.8.3/lib/httpclient.rb:1014:in `do_request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/httpclient-2.8.3/lib/httpclient.rb:856:in `request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/faraday-0.9.2/lib/faraday/adapter/httpclient.rb:36:in `call'
         # ./lib/common/client/middleware/request/remove_cookies.rb:11:in `call'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_response'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/faraday-0.9.2/lib/faraday/connection.rb:377:in `run_request'
         # /home/kf/.rvm/gems/ruby-2.4.5@vets-api/gems/faraday-0.9.2/lib/faraday/connection.rb:140:in `get'
         # ./lib/common/client/base.rb:91:in `request'
         # ./spec/lib/common/client/concerns/monitoring_spec.rb:31:in `block in request'
         # ./lib/common/client/concerns/monitoring.rb:9:in `with_monitoring'
         # ./spec/lib/common/client/concerns/monitoring_spec.rb:31:in `request'
         # ./spec/lib/common/client/base_spec.rb:26:in `block (4 levels) in <top (required)>'
         # ./spec/lib/common/client/base_spec.rb:26:in `block (3 levels) in <top (required)>'
cookies.rb:11:ient.cookie_mab:50 :/client/base_spspec/lib/common/client/
Testraise SecurityError, 'http client needs cookies stripped'
..................FF................Common::Client::SecurityError:
       http client needs cookies stripped
 Common::Client::Errors::ClientError, got #<Common::Client::SecurityError: http client needs cookies stripped> with backtrace:
         # ./lib/common/client/base.rb:70:in `block in connection'
e cookies stripped
 cookies sNotImplementeFaraday::Adapter::HTTPClient:Clien  # ./spec/lib/common/client/base_spec.rb:26:in `block (3 levels) in <top (required)>'

 -S+  statsd_roster:
+    namespace: statsd-roster
+    each_ttl: 604800
v0/gi_bill_feedbacksPMS-d s 8827:C 17 Sep 2019 12:56:34.456 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 12:56:34.526 * Background saving terminated with success
896:M 17 Sep 2019 13:01:35.063 * 10 changes in 300 szrange" "breakers:Search/Results-outages" "-1" "-1"
1568743004.727600 [0 [::1]:48528] "multi"
1568743004.727693 [0 [::1]:48528] "incr" "breakers:Search/Results-successes-1568742960"
1568743004.727709 [0 [::1]:48528] "expire" "breakers:Search/Results-successes-1568742960" "2592000"
1568743004.727721 [0 [::1]:48528] "exec"
1568743004.733178 [0 [::1]:35888] "set" "statsd-roster:api.search.results.total" "{\":key\":\"api.search.results.total\"}"
1568743004.733493 [0 [::1]:35888] "expire" "statsd-roster:api.search.results.total" "604800"
1568743005.878088 [0 [::1]:48528] "zrange" "breakers:Search/Results-outages" "-1" "-1"
1568743005.880967 [0 [::1]:48528] "multi"
127.0.0.1:6379> monitor
exec"
"set" "statsd-roster:api.search.results.total" "{\":key\":\"api.search.results.total\"}"
 "{\":key\":\"api.search.results.total\"}"
statsd-roster:api.search.results.total"statsd-roster:api.search.results.total" "604800"
::1]:48528] "exec"
1568743807.142566 [0 [::1]:48528] "zadd" "breakers:PPMS-outages" "1568743807" "{\"start_time\":1568743807,\"forced\":false}"
1568743807.149860 [0 [::1]:48528] "multi"
1568743807.150446 [0 [::1]:48528] "zrem" "breakers:PPMS-outages" "{\"start_time\":1568743379,\"forced\":false}"
1568743807.150474 [0 [::1]:48528] "zadd" "breakers:PPMS-outages" "1568743379" "{\"start_time\":1568743379,\"forced\":false,\"last_test_time\":1568743807}"
1568743807.150796 [0 [::1]:48528] "exec"
6:M 17 Sep 2019 12:56:34.426 * Background saving started by pid 8827
8827:C 17 Sep 2019 12:56:34.454 * DB saved on disk
8827:C 17 Sep 2019 12:56:34.456 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 12:56:34.526 * Background saving terminated with success
896:M 17 Sep 2019 13:01:35.063 * 10 changes in 300 seconds. Saving...
896:M 17 Sep 2019 13:01:35.064 * Background saving started by pid 30986
30986:C 17 Sep 2019 13:01:35.077 * DB saved on disk
30986:C 17 Sep 2019 13:01:35.081 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:01:35.165 * Background saving terminated with success
896:M 17 Sep 2019 13:09:20.499 * 10 changes in 300 seconds. Saving...
896:M 17 Sep 2019 13:09:20.501 * Background saving started by pid 2349
2349:C 17 Sep 2019 13:09:20.527 * DB saved on disk
2349:C 17 Sep 2019 13:09:20.528 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:09:20.601 * Background saving terminated with success
896:M 17 Sep 2019 13:14:21.095 * 10 changes in 300 seconds. Saving...
896:M 17 Sep 2019 13:14:21.095 * Background saving started by pid 28190
28190:C 17 Sep 2019 13:14:21.106 * DB saved on disk
28190:C 17 Sep 2019 13:14:21.107 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:14:21.196 * Background saving terminated with success
896:M 17 Sep 2019 13:16:14.293 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 13:16:14.294 * Background saving started by pid 3552
3552:C 17 Sep 2019 13:16:14.314 * DB saved on disk
3552:C 17 Sep 2019 13:16:14.316 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:16:14.394 * Background saving terminated with success
896:M 17 Sep 2019 13:17:49.838 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 13:17:49.839 * Background saving started by pid 9460
9460:C 17 Sep 2019 13:17:49.846 * DB saved on disk
9460:C 17 Sep 2019 13:17:49.848 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:17:49.939 * Background saving terminated with success
896:M 17 Sep 2019 13:21:24.718 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 13:21:24.718 * Background saving started by pid 24555
24555:C 17 Sep 2019 13:21:24.730 * DB saved on disk
24555:C 17 Sep 2019 13:21:24.731 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:21:24.818 * Background saving terminated with success
896:M 17 Sep 2019 13:25:01.389 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 13:25:01.389 * Background saving started by pid 6277
6277:C 17 Sep 2019 13:25:01.402 * DB saved on disk
6277:C 17 Sep 2019 13:25:01.404 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:25:01.489 * Background saving terminated with success
896:M 17 Sep 2019 13:26:24.407 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 13:26:24.408 * Background saving started by pid 11870
11870:C 17 Sep 2019 13:26:24.418 * DB saved on disk
11870:C 17 Sep 2019 13:26:24.420 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:26:24.509 * Background saving terminated with success
896:M 17 Sep 2019 13:27:37.162 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 13:27:37.163 * Background saving started by pid 16452
16452:C 17 Sep 2019 13:27:37.173 * DB saved on disk
16452:C 17 Sep 2019 13:27:37.175 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:27:37.263 * Background saving terminated with success
896:M 17 Sep 2019 13:32:38.044 * 10 changes in 300 seconds. Saving...
896:M 17 Sep 2019 13:32:38.045 * Background saving started by pid 2699
2699:C 17 Sep 2019 13:32:38.053 * DB saved on disk
2699:C 17 Sep 2019 13:32:38.054 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:32:38.145 * Background saving terminated with success
896:M 17 Sep 2019 14:00:17.079 * 1 changes in 900 seconds. Saving...
896:M 17 Sep 2019 14:00:17.080 * Background saving started by pid 7559
7559:C 17 Sep 2019 14:00:17.088 * DB saved on disk
7559:C 17 Sep 2019 14:00:17.090 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 14:00:17.180 * Background saving terminated with success
896:M 17 Sep 2019 14:07:52.114 * 10 changes in 300 seconds. Saving...
896:M 17 Sep 2019 14:07:52.115 * Background saving started by pid 4290
4290:C 17 Sep 2019 14:07:52.123 * DB saved on disk
4290:C 17 Sep 2019 14:07:52.124 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 14:07:52.215 * Background saving terminated with success
896:M 17 Sep 2019 14:09:32.859 * 10000 changes in 60 seconds. Saving...
896:M 17 Sep 2019 14:09:32.860 * Background saving started by pid 9906
9906:C 17 Sep 2019 14:09:32.868 * DB saved on disk
9906:C 17 Sep 2019 14:09:32.869 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 14:09:32.960 * Background saving terminated with success

896:M 17 Sep 2019 13:27:37.163 * Background saving started by pid 16452
16452:C 17 Sep 2019 13:27:37.173 * DB saved on disk
16452:C 17 Sep 2019 13:27:37.175 * RDB: 2 MB of memory used by copy-on-write
896:M 17 Sep 2019 13:27:37.263 * Background saving terminated with success
896:M 17 Sep 2019 13:32:38.044 * 10 changes in 300 seconds. Saving...
896:M 17 Sep 2019 13:32:38.045 * Background saving started by pid 2699
2699:C 17 Sep 2019 13:32:38.053 * DB saved on disk
2699:C 17 Sep 2019 13:32:38.054 * RDB: 2 MB of memory used by copy-on-write
14:32:18 oddball/department-of-veterans-affairs/vets-api  master [!$]    ruby-2.4.5@vets-api
➔ ab -n 3000 -c 50 "staging-api.va.gov/v0/search?query=benefits"
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking staging-api.va.gov (be patient)
Completed 300 requests
Completed 600 requests
Completed 900 requests
Completed 1200 requests
Completed 1500 requests
Completed 1800 requests
Completed 2100 requests
Completed 2400 requests
Completed 2700 requests
Completed 3000 requests
Finished 3000 requests


Server Software:
Server Hostname:        staging-api.va.gov
Server Port:            80

Document Path:          /v0/search?query=benefits
Document Length:        784 bytes

Concurrency Level:      50
Time taken for tests:   42.413 seconds
Complete requests:      3000
Failed requests:        0
Total transferred:      2778000 bytes
HTML transferred:       2352000 bytes
Requests per second:    70.73 [#/sec] (mean)
Time per request:       706.885 [ms] (mean)
Time per request:       14.138 [ms] (mean, across all concurrent requests)
Transfer rate:          63.96 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:       70  396 376.0    355    9155
Processing:    61  297 243.3    252    3131
Waiting:       60  287 213.5    251    3131
Total:        132  692 436.0    622    9355

Percentage of the requests served within a certain time (ms)
  50%    622
  66%    672
  75%    707
  80%    724
  90%    795
  95%   1189
  98%   1882
  99%   3484
 100%   9355 (longest request)

14:33:13 oddball/department-of-veterans-affairs/vets-api  master [!$]    ruby-2.4.5@vets-api
➔ ab -n 3000 -c 50 "staging-api.va.gov/v0/search?query=benefits"
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking staging-api.va.gov (be patient)
Completed 300 requests
Completed 600 requests
Completed 900 requests
Completed 1200 requests
Completed 1500 requests
Completed 1800 requests
Completed 2100 requests
Completed 2400 requests
Completed 2700 requests
Completed 3000 requests
Finished 3000 requests


Server Software:
Server Hostname:        staging-api.va.gov
Server Port:            80

Document Path:          /v0/search?query=benefits
Document Length:        784 bytes

Concurrency Level:      50
Time taken for tests:   45.536 seconds
Complete requests:      3000
Failed requests:        0
Total transferred:      2778000 bytes
HTML transferred:       2352000 bytes
Requests per second:    65.88 [#/sec] (mean)
Time per request:       758.928 [ms] (mean)
Time per request:       15.179 [ms] (mean, across all concurrent requests)
Transfer rate:          59.58 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:       52  390 354.0    341    3396
Processing:    39  310 273.0    257    6982
Waiting:       39  297 240.5    255    6982
Total:         95  700 445.8    638   10267

Percentage of the requests served within a certain time (ms)
  50%    638
  66%    680
  75%    701
  80%    708
  90%    773
  95%   1117
  98%   1894
  99%   3490
 100%  10267 (longest request)

14:34:50 oddball/department-of-veterans-affairs/vets-api  master [!$]    ruby-2.4.5@vets-api
➔ ab -n 3000 -c 50 "staging-api.va.gov/v0/search?query=benefits"
This is ApacheBench, Version 2.3 <$Revision: 1843412 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking staging-api.va.gov (be patient)
Completed 300 requests
Completed 600 requests
Completed 900 requests
Completed 1200 requests
Completed 1500 requests
Completed 1800 requests
Completed 2100 requests
Completed 2400 requests
Completed 2700 requests
Completed 3000 requests
Finished 3000 requests


Server Software:
Server Hostname:        staging-api.va.gov
Server Port:            80

Document Path:          /v0/search?query=benefits
Document Length:        784 bytes

Concurrency Level:      50
Time taken for tests:   41.881 seconds
Complete requests:      3000
Failed requests:        0
Total transferred:      2778000 bytes
HTML transferred:       2352000 bytes
Requests per second:    71.63 [#/sec] (mean)
Time per request:       698.012 [ms] (mean)
Time per request:       13.960 [ms] (mean, across all concurrent requests)
Transfer rate:          64.78 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:       48  409 351.2    373    9194
Processing:    44  281 316.0    256    7855
Waiting:       43  273 296.8    252    7855
Total:         92  690 509.2    635   11419

Percentage of the requests served within a certain time (ms)
  50%    635
  66%    666
  75%    691
  80%    712
  90%    784
  95%    894
  98%   1592
  99%   3429

```

These results are against staging/master before merge

**Pull request merged**